### DISCUSSION TITLE
Authentication enabled with Sunbird RC

### DISCUSSION THREAD

**Discussion by sagarihanda** [Sunbird-Lern/Community#29] [Query]
**Context**
UP State Medical Faculty (SMF) is building a platform to manage human resources for allied healthcare professionals (nurses, paramedical staff, lab technicians etc.) in the state. The Government is looking to build the platform on Sunbird building blocks to adhere to ABDM's (https://abdm.gov.in/abdm) principles. One of the key modules being built is to enable competency based assessments of allied professionals, and to enable mid-career learning courses via an LMS. Six Sunbird blocks (RC, Obsrv, Knowlg, inQuiry, CoKreat and Ed) are being leveraged to develop the platform with the user data being housed in Sunbird RC. 

**Clarification**
Currently, the authentication for Lern is coupled with DIKSHA log-in. We need to decouple it and link it to Sunbird RC. Is this planned in the current building block roadmap? And if so, what timeline can we expect? 

@rahul101001000

**Comment by rhwarrier** [comment]
Thanks for reaching out. Sunbird login currently uses a KeyCloak module to authenticate users. User details are stored in a user table that is part of the Lern building block. Is the suggestion to be able to use SB RC in place of this?
Please clarify the use-case a little better so we can understand the requirements clearly. 
As of now, Lern uses RC only for credentials, and there is no other roadmap item planned. 
cc : @reshmi-nair  @vrayulu

**Comment by rahul101001000** [comment]
Thanks @rhwarrier 

Yes exactly, we would like to use Sunbird Lern but with Sunbird RC for auth and user management. Any way this can be done sooner or you could guide us on how to do it? 

Our backup here is to do an oauth SSO like we have done in Himachal with DIKSHA, but would really like to use Sunbird RC since it would power other things in the platform. 

@pramodkvarma

**Reply by rhwarrier** [reply]
This will need further discussion. Currently, there are multiple tables related to the user that are referenced internally for various actions. Also, we'll have to ensure performance is not affected (the User APIs support high TPS)
 @vrayulu - any thoughts on how to take this forward ?

Let's set up some time to discuss this so that the details can be ironed out.
We could also look at doing a minimal switch over based on current needs. Event if it means not porting the full functionality across using RC for User management.  (assuming such an implementation is possible)

@reshmi-nair  requesting you to please set up some time with the team and  @vrayulu to discuss this further.
Thanks.

**Comment by rahul101001000** [comment]
Thanks @rhwarrier, yes all fair points to consider. btw just cleanly allowing for a third party authentication and authorization to interface with Lern would also help a lot. That way it is decoupled and the other side could be Sunbird-RC or a state system.

Please do include @ChakshuGautam @sagarihanda and me on the invite. 

Thanks!

**Reply by pramodkvarma** [reply]
As I understand, Lern uses a standard interface to deal with authentication (behind the scene it could be any ID and SSO/oAuth provider). But, even within this if we need to abstract and decouple, sure, let us do that.

**Reply by reshmi-nair** [reply]
@rahul101001000 @rhwarrier  I have setup a meeting (on Aug 3rd - 2.30pm) for discussing the details.
